src/containers/RequestBuilder.js

```diff
@@ -169,197 +175,193 @@ export default class RequestBuilder extends Component {
         method: 'POST',
         headers: new Headers(headers),
         body: JSON.stringify(json_request),
-        signal: this.timeout(10).signal //Timeout set to 10 seconds
+        signal: timeout(10).signal //Timeout set to 10 seconds
       })
         .then(response => {
-          clearTimeout(this.timeout);
+          clearTimeout(timeout);
           response.json().then(fhirResponse => {
             console.log(fhirResponse);
             if (fhirResponse?.status) {
-              this.consoleLog(
+              consoleLog(
                 'Server returned status ' + fhirResponse.status + ': ' + fhirResponse.error,
                 types.error
               );
-              this.consoleLog(fhirResponse.message, types.error);
+              consoleLog(fhirResponse.message, types.error);
             } else {
-              this.setState({ response: fhirResponse });
+              setState(prevState => ({ ...prevState, response: fhirResponse }));
             }
-            this.setState({ loading: false });
+            setState(prevState => ({ ...prevState, loading: false }));
           });
         })
         .catch(() => {
-          this.consoleLog('No response received from the server', types.error);
-          this.setState({ response: {} });
-          this.setState({ loading: false });
+          consoleLog('No response received from the server', types.error);
+          setState(prevState => ({ ...prevState, response: {}, loading: false }));
         });
     } catch (error) {
-      this.setState({ loading: false });
-      this.consoleLog('Unexpected error occurred', types.error);
+      
+      setState(prevState => ({ ...prevState, loading: false }));
+      consoleLog('Unexpected error occurred', types.error);
       if (error instanceof TypeError) {
-        this.consoleLog(error.name + ': ' + error.message, types.error);
+        consoleLog(error.name + ': ' + error.message, types.error);
       }
-      this.setState({ loading: false });
     }
-  }
+  };
 
-  takeSuggestion(resource) {
+  const takeSuggestion = (resource) => {
     // when a suggestion is taken, call into the requestBox to resubmit the CRD request with the new request
-    this.requestBox.current.replaceRequestAndSubmit(resource);
-  }
+    requestBox.current.replaceRequestAndSubmit(resource);
+  };
 
-  getPatients = () => {
-    if (this.props.globalState.patientFhirQuery) {
-      this.props.client
-        .request(this.props.globalState.patientFhirQuery, { flat: true })
+  const getPatients = () => {
```

Comment:
one can technically circumvent the issue by changing the patient FHIR query, clicking the save button, and then either clicking Reconnect EHR or refreshing the page and then the saved patient FHIR query will get loaded in (and you'll see the desired # patients)

Reply:
weirdly enough, if I sometimes click Save after modifying the patient FHIR query field, if I 
* change the field from 2 to 12, and go back to the refresh button the page won't crash (and the number of patients fetched changes)
* or if I do it again but with 2 to 1 it _does_ crash

I tried reproducing this again with some more specific numbers:
* 12 to 23 (no crash), 23 to 12 (no crash), 12 to 1 (crash)
* 12 to 23 (no crash), 23 to 1 (crash)

when it doesn't crash this is probably because the currently rendered number of fetched patients (which is 3, and the maximum amount we have anyways) stays the same between both renders

Reply:
Yeah I discovered it is crashing when the patient box is open and then you try to refresh, I have it setting expanded: false before it makes the new query, then opening it back up after. This no longer causes the bug.